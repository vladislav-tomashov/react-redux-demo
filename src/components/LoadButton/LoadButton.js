import React from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { loadCurrencyRates } from "../../store/actions/currencyRates/currencyRatesActionCreators";
import {
  isCurrencyRatesLoading,
  wasCurrencyRatesLoaded
} from "../../store/selectors/currencyRatesSelectors";
import "./LoadButton.scss";

const LoadButton = ({
  loading = false,
  loaded = false,
  loadCurrencyRates = () => {}
} = {}) => {
  let label = "Load rates";
  if (loading) {
    label = "Loading...";
  } else if (loaded) {
    label = "Update rates";
  }
  return (
    <button
      className="LoadButton"
      disabled={loading}
      onClick={loadCurrencyRates}
    >
      {label}
    </button>
  );
};

LoadButton.propTypes = {
  loadCurrencyRates: PropTypes.func,
  loading: PropTypes.bool,
  loaded: PropTypes.bool
};

const mapStateToProps = (state /*, ownProps*/) => {
  return {
    loading: isCurrencyRatesLoading(state),
    loaded: wasCurrencyRatesLoaded(state)
  };
};

const mapDispatchToProps = dispatch =>
  bindActionCreators({ loadCurrencyRates }, dispatch);

const ConnectedLoadButton = connect(
  mapStateToProps,
  mapDispatchToProps
)(LoadButton);

export { ConnectedLoadButton as default, LoadButton };
