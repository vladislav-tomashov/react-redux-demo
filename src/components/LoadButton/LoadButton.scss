@import "../../styles/settings";

.LoadButton {
  background: $blue;
  border: 1px solid $off-white;
  color: $off-white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}
