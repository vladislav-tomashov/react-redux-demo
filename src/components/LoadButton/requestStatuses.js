export const LOADING = "loading";
export const LOADED = "loaded";
export const NOTDONE = "notDone";
