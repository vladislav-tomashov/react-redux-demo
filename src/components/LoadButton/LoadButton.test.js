import React from "react";
import { shallow } from "enzyme";
import { LoadButton } from "./LoadButton";

let wrapper;

describe("LoadButton component", () => {
  beforeEach(() => {
    wrapper = shallow(<LoadButton />);
  });

  test("should render LoadButton correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render LoadButton with 'loading' props correctly", () => {
    wrapper.setProps({
      loading: true
    });
    expect(wrapper).toMatchSnapshot();
  });
  test("should render LoadButton with 'loaded' props correctly", () => {
    wrapper.setProps({
      loaded: true
    });
    expect(wrapper).toMatchSnapshot();
  });
  test("should render LoadButton with all props correctly", () => {
    const loadCurrencyRates = jest.fn();
    wrapper.setProps({
      loading: false,
      loaded: true,
      loadCurrencyRates
    });
    expect(wrapper).toMatchSnapshot();
  });
  test("should handle button click", () => {
    const loadCurrencyRates = jest.fn();
    wrapper.setProps({
      loading: false,
      loaded: true,
      loadCurrencyRates
    });
    wrapper.find("button").simulate("click");
    expect(loadCurrencyRates).toBeCalledTimes(1);
  });
});
