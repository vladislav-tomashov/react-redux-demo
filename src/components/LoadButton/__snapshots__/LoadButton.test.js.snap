// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadButton component should render LoadButton correctly 1`] = `
<button
  className="LoadButton"
  disabled={false}
  onClick={[Function]}
>
  Load rates
</button>
`;

exports[`LoadButton component should render LoadButton with 'loaded' props correctly 1`] = `
<button
  className="LoadButton"
  disabled={false}
  onClick={[Function]}
>
  Update rates
</button>
`;

exports[`LoadButton component should render LoadButton with 'loading' props correctly 1`] = `
<button
  className="LoadButton"
  disabled={true}
  onClick={[Function]}
>
  Loading...
</button>
`;

exports[`LoadButton component should render LoadButton with all props correctly 1`] = `
<button
  className="LoadButton"
  disabled={false}
  onClick={[MockFunction]}
>
  Update rates
</button>
`;
